src/components/views/elements/AppTile.js

```diff
@@ -98,43 +118,73 @@ export default React.createClass({
         return false;
     },
 
-    componentWillMount: function() {
-        if (!this.isScalarUrl()) {
+    componentWillMount() {
+        window.addEventListener('message', this._onMessage, false);
+        this.setScalarToken();
+    },
+
+    /**
+     * Adds a scalar token to the widget URL, if required
+     * Component initialisation is only complete when this function has resolved
+     */
+    setScalarToken() {
+        this.setState({initialising: true});
```

Comment:
Could do this immediately preceding the getScalarToken? Then you don't have to do the initialising: false on line 138.

Reply:
You do still need to always set initialising to false, as it is **always** set to true in the initialiser. Leaving this as is.

Reply:
Oh, that's true.